ledger/sandbox/src/main/scala/com/digitalasset/platform/sandboxnext/Runner.scala

```diff
@@ -0,0 +1,184 @@
+import akka.stream.Materializer
+import com.codahale.metrics.SharedMetricRegistries
+import com.daml.ledger.on.sql.Database.InvalidDatabaseException
+import com.daml.ledger.on.sql.SqlLedgerReaderWriter
+import com.daml.ledger.participant.state.kvutils.api.KeyValueParticipantState
+import com.daml.ledger.participant.state.v1
+import com.daml.ledger.participant.state.v1.{ReadService, WriteService}
+import com.digitalasset.api.util.TimeProvider
+import com.digitalasset.daml.lf.archive.DarReader
+import com.digitalasset.daml.lf.data.Ref
+import com.digitalasset.daml_lf_dev.DamlLf.Archive
+import com.digitalasset.ledger.api.auth.{AuthService, AuthServiceWildcard}
+import com.digitalasset.logging.LoggingContext.newLoggingContext
+import com.digitalasset.logging.{ContextualizedLogger, LoggingContext}
+import com.digitalasset.platform.apiserver.{ApiServerConfig, StandaloneApiServer}
+import com.digitalasset.platform.common.LedgerIdMode
+import com.digitalasset.platform.configuration.BuildInfo
+import com.digitalasset.platform.indexer.{
+  IndexerConfig,
+  IndexerStartupMode,
+  StandaloneIndexerServer
+}
+import com.digitalasset.platform.sandbox.banner.Banner
+import com.digitalasset.platform.sandbox.config.SandboxConfig
+import com.digitalasset.platform.sandboxnext.Runner._
+import com.digitalasset.resources.ResourceOwner
+import com.digitalasset.resources.akka.AkkaResourceOwner
+import scalaz.syntax.tag._
+
+import scala.compat.java8.FutureConverters.CompletionStageOps
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Try
+
+/**
+  * Runs Sandbox with a KV SQL ledger backend.
+  *
+  * Known issues:
+  *   - does not support authorization
+  *   - does not support static time
+  *   - does not provide the reset service
+  */
+class Runner {
+  def owner(config: SandboxConfig): ResourceOwner[Unit] = {
+    implicit val system: ActorSystem = ActorSystem("sandbox")
+    implicit val materializer: Materializer = Materializer(system)
+    implicit val executionContext: ExecutionContext = system.dispatcher
+
+    val ledgerId: v1.LedgerId = Ref.LedgerString.assertFromString(config.ledgerIdMode match {
+      case LedgerIdMode.Static(ledgerId) => ledgerId.unwrap
+      case LedgerIdMode.Dynamic() => UUID.randomUUID.toString
+    })
+
+    val (ledgerType, ledgerJdbcUrl, indexJdbcUrl) = config.jdbcUrl match {
+      case Some(url) if url.startsWith("jdbc:postgresql") => ("PostgreSQL", url, url)
+      case Some(url) if url.startsWith("jdbc:h2:mem:") => ("in-memory", InMemoryLedgerJdbcUrl, url)
+      case Some(url) if url.startsWith("jdbc:h2:") =>
+        throw new InvalidDatabaseException(
+          "This version of Sandbox does not support file-based H2 databases. Please use SQLite instead.")
+      case Some(url) if url.startsWith("jdbc:sqlite:") => ("SQLite", url, InMemoryIndexJdbcUrl)
+      case None => ("in-memory", InMemoryLedgerJdbcUrl, InMemoryIndexJdbcUrl)
+    }
+
+    newLoggingContext { implicit logCtx =>
```

Comment:
I'm not sure we need the logging context to wrap the whole thing. I would suggest you restrict the scope to where it's actually used. Can you check where this is needed? From a cursory look you can probably move it inside the `startParticipant` method (and avoid asking for an `implicit` one).

Reply:
That's where it was when I copied it from the KVUtils _Runner_, but it's also needed for the `SqlLedgerReaderWriter`.